application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -107,68 +110,71 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), userGuildId);
         if (modMailAuditLog.isEmpty()) {
             logger.warn(
-                    "Cannot find the designated modmail channel in server by id {} with the pattern modmail",
-                    userGuildId);
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    userGuildId, configModMailChannelPattern);
             return;
         }
 
-        MessageCreateAction message = createMessage(event, userId, modMailAuditLog.orElseThrow());
+        MessageCreateAction message =
+                createModMessage(event, userId, modMailAuditLog.orElseThrow());
 
         sendMessage(event, message);
 
     }
 
     private boolean handleIsOnCooldown(long userId, SlashCommandInteractionEvent event) {
-        if (isChannelOnCooldown(userId)) {
-            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
-                .setEphemeral(true)
-                .queue();
-            return true;
+        if (!isChannelOnCooldown(userId)) {
+            return false;
         }
-        return false;
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
     }
 
-    private Optional<TextChannel> getModMailChannel(JDA jda, long userGuildI) {
-        return jda.getGuildById(userGuildI)
+    private Optional<TextChannel> getModMailChannel(JDA jda, long guildId) {
+        return jda.getGuildById(guildId)
             .getTextChannelCache()
             .stream()
             .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
             .findAny();
     }
 
-    private MessageCreateAction createMessage(SlashCommandInteractionEvent event, long userId,
+    private MessageCreateAction createModMessage(SlashCommandInteractionEvent event, long userId,
             TextChannel modMailAuditLog) {
         String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
         boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
-        String user;
-        MessageCreateAction message;
-        if (wantsToStayAnonymous) {
-            user = "Anonymous";
-            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage));
-        } else {
-            user = event.getUser().getAsMention();
-            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage))
-                .addActionRow(DiscordClientAction.General.USER.asLinkButton("Author Profile",
-                        String.valueOf(userId)));
+
+        User user = event.getUser();
+        MessageCreateAction message =
+                modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage));
```

Comment:
ur supposed to give the method null if the user wants to be anonymous. are u doing this? doesnt look like

Reply:
Weirdly I thought that `createModMailMessage` was the one solving that , now that I see it better, it actually just builds on it 😅